src/app/sharehouses/page.tsx

```diff
@@ -1,32 +1,32 @@
+import Link from 'next/link';
 import { Progress } from '@/components/ui/piechart';
 import { LandlordDashboard } from '@/components/landlord-dashboard/LandlordDashboard';
-import Link from 'next/link';
 import { FloatingActionButton } from '@/components/ui/floatingActionButton';
+import { api } from '@/lib/hono';
+import { headers } from 'next/headers';
+import { CONSTRAINTS } from '@/constants/constraints';
 
-interface IShareHouse {
-  shareHouseId: string;
-  shareHouseName: string;
-  progressPercent: number;
-}
-
-const ShareHousesPage = () => {
-  const shareHouses: IShareHouse[] = [
+const ShareHousesPage = async () => {
+  // Fetch share house information with headers including cookies
+  const res = await api.sharehouses.$get(
+    {},
     {
-      shareHouseId: 'yourShareHouseId',
-      shareHouseName: 'Sample Share House',
-      progressPercent: 50,
+      headers: {
+        cookie: headers().get('cookie') || '', // Add cookies to headers
+      },
     },
-    {
-      shareHouseId: 'yourShareHouseId2',
-      shareHouseName: 'Sample Share House 2',
-      progressPercent: 80,
-    },
-    {
-      shareHouseId: 'yourShareHouseId3',
-      shareHouseName: 'Sample Share House 3',
-      progressPercent: 20,
-    },
-  ];
+  );
+
+  // Convert response to JSON
+  const data = await res.json();
+
+  // Check for error in data and display it if found
+  if ('error' in data) {
+    return <div>{data.error}</div>;
```

Comment:
This is temporary, right? Then, could you add a TODO comment so that it can be noticed to fix it later?

Reply:
Thank you for reviewing! I added a TODO comment.